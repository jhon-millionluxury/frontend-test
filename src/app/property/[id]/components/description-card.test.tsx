import { render, screen } from "@testing-library/react";
import PropertyDetailPageDescriptionCard from "@/app/property/[id]/components/description-card";

describe("PropertyDetailPageDescriptionCard", () => {
  it("renders the heading and description correctly", () => {
    const description = "This is a beautiful property with a garden.";

    render(<PropertyDetailPageDescriptionCard description={description} />);

    expect(screen.getByText("About This Property")).toBeInTheDocument();

    expect(screen.getByText(description)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <PropertyDetailPageDescriptionCard description="Snapshot description" />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
