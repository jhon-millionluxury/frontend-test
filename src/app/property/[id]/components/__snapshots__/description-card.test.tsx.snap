// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PropertyDetailPageDescriptionCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-card text-card-foreground flex flex-col gap-6 rounded-xl border py-6 shadow-sm mb-8"
    data-slot="card"
  >
    <div
      class="p-6"
      data-slot="card-content"
    >
      <h2
        class="mb-4 font-serif text-2xl font-semibold text-foreground"
      >
        About This Property
      </h2>
      <p
        class="leading-relaxed text-muted-foreground"
      >
        Snapshot description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
